### PR TITLE
Update Check points Rutanio

### PR DESCRIPTION
**Pull request by turcol**

### CODE REVIEW

**Review by dangershony — APPROVED** [review]

**Comment by dangershony** [comment]
ah build fails

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs [hunk 1]
```diff
@@ -698,13 +699,18 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
...
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Hex OpReturn error.", "The OpReturnData is set as a hex value and could not be decoded");
+
                 var context = new TransactionBuildContext(this.network)
                 {
                     AccountReference = new WalletAccountReference(request.WalletName, request.AccountName),
                     TransactionFee = string.IsNullOrEmpty(request.FeeAmount) ? null : Money.Parse(request.FeeAmount),
                     MinConfirmations = request.AllowUnconfirmed ? 0 : 1,
                     Shuffle = request.ShuffleOutputs ?? true, // We shuffle transaction outputs by default as it's better for anonymity.
                     OpReturnData = request.OpReturnData,
+                    OpReturnRawData = opReturnRawData,
```
Comment (sondreb) [inline src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:713]: What is the difference between OpReturnData and OpReturnRawData? Can't the consumer simply parse OpReturnData to Hex if they need to?

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs [hunk 2]
```diff
@@ -1602,5 +1608,17 @@ private void SyncFromBestHeightForRecoveredWallets(DateTime walletCreationDate)
                 this.walletSyncManager.SyncFromHeight(blockHeightToSyncFrom);
             }
         }
+
+        private (byte[], bool) TryGetHexValue(string hexString)
```
Comment (sondreb) [inline src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:1612]: This is probably not the place for this method, don't we have an utility somewhere for this?

**Review by sondreb — CHANGES_REQUESTED** [review]

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs (hunk 1 above)
Comment (turcol) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:713]: The OpReturnData is an ASCII (String) and OpReturnRawData is Bytes

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs (hunk 2 above)
Comment (turcol) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:1612]: @sondreb where should we place it? We have it there, as there was another method like this [SyncFromBestHeightForRecoveredWallets]

Comment (sondreb) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:1612]: I had a look at the method now and it just redirect the call to another utility library, so I guess it's fine =)

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs (hunk 1 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:713]: It is the same data in different formats, I don't think the API should be responsible for returning the same data in multiple formats. Why can't the transformation happen by the API consumer?

Comment (turcol) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:713]: In this case you can send data to the API (for the OP_RETURN) in any format, HEX or ASCII..  more options (easy for the API consumer to use it)

p.e: you can easily send a hash like this, and will be 32 bytes and not 64 ASCII chars
35544c02f4cd4f403fab7579c6385400dbab656558b1d9750337ddb9f844e229

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs [hunk 3]
```diff
@@ -698,13 +699,18 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
                     }
                 }
 
+                var (opReturnRawData, isValid) = request.OpReturnDataIsHex ? TryGetHexValue(request.OpReturnData) : (null, true);
```
Comment (sondreb) [inline src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:702]: If this is to be accepted, it must also be added to the "GetTransactionFeeEstimate" method or the fee calculation will be wrong and different compared to the actual transaction building.

**Review by sondreb — CHANGES_REQUESTED** [review]

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs (hunk 1 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:713]: Reviewed this and this is acceptable, see the other issue about duplicating the same logic in the fee calculation method.

File: src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs (hunk 3 above)
Comment (sondreb) [reply src/Features/Blockcore.Features.Wallet/Api/Controllers/WalletController.cs:702]: I'm fixing this myself, adding the same logic to the fee-calculation method.

**Review by sondreb — APPROVED** [review]
This can now be approved.

**Pull request merged**